src/config.ts

```diff
@@ -159,14 +162,20 @@ export function defaultMeasurement(url: LocalUrl|RemoteUrl): Measurement {
  * Validate the given JSON object parsed from a config file, and expand it into
  * a fully specified configuration.
  */
-export async function parseConfigFile(parsedJson: unknown): Promise<Config> {
+export async function parseConfigFile(
+    parsedJson: unknown, configFilename?: string): Promise<Config> {
   const schema = require('./config.schema.json');
   const result =
       jsonschema.validate(parsedJson, schema, {propertyName: 'config'});
   if (result.errors.length > 0) {
     throw new Error(result.errors[0].toString());
   }
   const validated = parsedJson as ConfigFile;
+  if (!('$schema' in validated) && configFilename) {
```

Comment:
Think I'd rather this was a separate function we call after parsing, since I wouldn't usually expect a parse function to mutate a file.